test/e2e/proxy_test.go

```diff
@@ -68,7 +68,8 @@ func TestSnowKubernetes121UbuntuProxyConfig(t *testing.T) {
 		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube121)),
 		framework.WithClusterFiller(api.WithControlPlaneCount(1)),
 		framework.WithClusterFiller(api.WithWorkerNodeCount(1)),
-		framework.WithProxy(),
+		// TODO: provide separate Proxy Env Vars for Snow provider. Leaving VSphere for backwards compatibility
+		framework.WithProxy(framework.VsphereProxyRequiredEnvVars),
```

Comment:
Actually if snow works, then I don't see why cloudstack would not work with the proxy config. @jiayiwang7 we don't need wireguard connection to connect to proxy for snow as well?

Reply:
ohh interesting, because that would allow us to avoid provider specific proxies. But I guess it's not a huge deal to introduce new ones now